dlib/data_io/cifar.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2020  Davis E. King (davis@dlib.net)
+// License: Boost Software License   See LICENSE.txt for the full license.
+#ifndef DLIB_CIFAR_CPp_
+#define DLIB_CIFAR_CPp_
+
+#include "cifar.h"
+#include <fstream>
+
+// ----------------------------------------------------------------------------------------
+
+namespace dlib
+{
+    namespace impl
+    {
+        void load_cifar_10_batch (
+            const std::string& folder_name,
+            const std::string& batch_name,
+            const size_t first_idx,
+            const size_t images_per_batch,
+            std::vector<matrix<rgb_pixel>>& images,
+            std::vector<unsigned long>& labels
+        )
+        {
+            std::ifstream fin((folder_name + "/" + batch_name).c_str(), std::ios::binary);
```

Comment:
Since C++11 the `.c_str()` is not needed anymore.

Reply:
I actually never used in my code, but saw that the MNIST code used it (I started late with C++, so not really aware of the pre C++11 era). Now I know, thank you :)